src/openforms/registrations/contrib/objects_api/registration_variables.py

```diff
@@ -94,6 +95,22 @@ def get_initial_value(self, submission: Submission | None = None):
         return submission.payments.get_completed_public_order_ids()
 
 
+@register("provider_payment_ids")
+class ProviderPaymentIds(BaseStaticVariable):
+    name = _("Provider payment IDs")
+    data_type = FormVariableDataTypes.array
+
+    def get_initial_value(self, submission: Submission | None = None):
+        if submission is None:
+            return None
+
+        return list(
+            submission.payments.filter(
+                status__in=(PaymentStatus.registered, PaymentStatus.completed)
+            ).values_list("provider_payment_id", flat=True)
```

Comment:
this `status__in` filter is repeated in quite a number of places now - perhaps we should add a queryset method `paid` to encapsulate it? So you can do `submission.payments.paid().values_list(...)`

Probably best to do this as a separate cleanup PR, so please create an issue for it (nice off-week task)

Reply:
https://github.com/open-formulieren/open-forms/issues/4588